include/Gaffer/SplinePlug.h

```diff
@@ -85,25 +95,23 @@ class SplinePlug : public ValuePlug
 		virtual bool acceptsChild( const GraphComponent *potentialChild ) const;
 		virtual PlugPtr createCounterpart( const std::string &name, Direction direction ) const;
 
-		const T &defaultValue() const;
+		const typename T::PointContainer &defaultPoints() const;
+		const Interpolation defaultInterpolation() const;
 		virtual void setToDefault();
 		virtual bool isSetToDefault() const;
 
-		/// Sets the value of all the child plugs by decomposing
-		/// the passed spline and storing it in the basis, points,
-		/// and endPointMultiplicity plug.
+
+		/// Sets the value of the points and interpolation child plugs
 		/// \undoable
-		void setValue( const T &value );
-		/// Recreates the spline by retrieving its basis and points
-		/// from the basis, points and endPointMultiplicity plugs.
-		T getValue() const;
-
-		ValuePlug *basisPlug();
-		const ValuePlug *basisPlug() const;
-		M44fPlug *basisMatrixPlug();
-		const M44fPlug *basisMatrixPlug() const;
-		IntPlug *basisStepPlug();
-		const IntPlug *basisStepPlug() const;
+		void setValue( const typename T::PointContainer &points, Interpolation interpolation );
+		/// Matching to setValue
+		void getValue( typename T::PointContainer &points, Interpolation &interpolation ) const;
+
+		T getValueAsCortexSpline() const;
```

Comment:
I think I would either just call this `spline()`, or possibly add it as a method on the `ValueType` I proposed, so you could do `plug->getValue().spline()`.